### PR TITLE
Adding TSG details for BA2024

### PR DESCRIPTION
**Pull request by marmegh**
Added TSG details for correcting BA2024.EnableSpectreMitigtations for static libraries

### CODE REVIEW

File: docs/RulesAndErrorsTroubleshootingGuide.md [hunk 1]
```diff
@@ -29,3 +29,19 @@ By default, BinSkim will only look for a PDB alongside the binary under analysis
...
+
+    warning BA2024: 'test.exe' was compiled with one or more modules that do not enable code generation mitigations for speculative execution side-channel attack (Spectre) vulnerabilities.
+
+### Resolving BA2024 warning
+
+Developers assume the security risk of static libraries that are linked into their binaries (because vulnerabilities in these dependencies may be exploitable once linked into other binaries).
+
+BA2024.EnableSpectreMitigations may fire against Microsoft default C++ static libraries, such as ‘LibCMT.lib’. In order to resolve these warnings, the developer must install and link to the appropriate MSVC Spectre-mitigated C++ runtime libraries. To do so:
+
+1. **Launch the ‘Visual Studio Installer’
```
Comment (eddynaka) [inline docs/RulesAndErrorsTroubleshootingGuide.md:45]: >** [](http://example.com/codeflow?start=3&length=2)

you could remove this

Comment (eddynaka) [inline docs/RulesAndErrorsTroubleshootingGuide.md:45]: >� [](http://example.com/codeflow?start=16&length=1)

replace this for `

Comment (eddynaka) [reply docs/RulesAndErrorsTroubleshootingGuide.md:45]: this is happening for 1/2/3 and in the line above.

Comment (eddynaka) [inline docs/RulesAndErrorsTroubleshootingGuide.md:45]: >� [](http://example.com/codeflow?start=40&length=1)

all phrases should end with a .

File: docs/RulesAndErrorsTroubleshootingGuide.md [hunk 2]
```diff
@@ -29,3 +29,19 @@ By default, BinSkim will only look for a PDB alongside the binary under analysis
...
+
+### Resolving BA2024 warning
+
+Developers assume the security risk of static libraries that are linked into their binaries (because vulnerabilities in these dependencies may be exploitable once linked into other binaries).
+
+BA2024.EnableSpectreMitigations may fire against Microsoft default C++ static libraries, such as `LibCMT.lib`. In order to resolve these warnings, the developer must install and link to the appropriate MSVC Spectre-mitigated C++ runtime libraries. To do so:
+
+1. Launch the `Visual Studio Installer`.
+2. Select the `Individual Components` tab and type `spectre` in the search field.
+3. Select and install Spectre-mitigated libs for your target platform.
```
Comment (michaelcfanning) [inline docs/RulesAndErrorsTroubleshootingGuide.md:47]: >platform [](http://example.com/codeflow?start=61&length=8)

This change is just fine as is. We could also try to generate a screenshot to help users, e.g. I actually haven't tried this.

Installing Spectre-mitigated C++ libraries in the VS Installer

File: docs/RulesAndErrorsTroubleshootingGuide.md [hunk 3]
```diff
@@ -29,3 +29,19 @@ By default, BinSkim will only look for a PDB alongside the binary under analysis
...
+
+    warning BA2024: 'test.exe' was compiled with one or more modules that do not enable code generation mitigations for speculative execution side-channel attack (Spectre) vulnerabilities.
+
+### Resolving BA2024 warning
+
+Developers assume the security risk of static libraries that are linked into their binaries (because vulnerabilities in these dependencies may be exploitable once linked into other binaries).
+
+BA2024.EnableSpectreMitigations may fire against Microsoft default C++ static libraries, such as `LibCMT.lib`. In order to resolve these warnings, the developer must install and link to the appropriate MSVC Spectre-mitigated C++ runtime libraries. To do so:
+
+1. Launch the `Visual Studio Installer`.
```
Comment (michaelcfanning) [inline docs/RulesAndErrorsTroubleshootingGuide.md:45]: >Launch [](http://example.com/codeflow?start=3&length=6)

Did you actually verify my guidance here to make sure it's accurate? This is always helpful testing. Looking at the steps, I think I may have missed something, you have to click 'Modify' to get to the individual components tab.

**Pull request closed without merging**